lib/portlayer/attach/attach.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package attach
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/lib/portlayer/constants"
+	"github.com/vmware/vic/lib/portlayer/exec"
+	"github.com/vmware/vic/pkg/trace"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// Handler defines the interface
+type Handler interface {
+	Join(handle interface{}) (interface{}, error)
+	Bind(handle interface{}) (interface{}, error)
+	Unbind(handle interface{}) (interface{}, error)
+}
+
+func toggle(h interface{}, connected bool) (interface{}, error) {
```

Comment:
This is not a part of the Port Layer API so could actually specify the type of the handle we require.